pkg/cmd/adm/register_member_test.go

```diff
@@ -459,47 +449,54 @@ func TestRegisterMember(t *testing.T) {
 		tcs := &toolchainv1alpha1.ToolchainClusterList{}
-		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-host-operator")))
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace(hostNs)))
 		assert.Empty(t, tcs.Items)
-		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-member-operator")))
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace(memberNs)))
 		assert.Empty(t, tcs.Items)
 	})
 }
 
-func mockCreateToolchainClusterInNamespaceWithReadyCondition(fakeClient *test.FakeClient, namespace string) {
+func mockCreateToolchainClusterInNamespaceWithReadyCondition(t *testing.T, fakeClient *test.FakeClient, namespace string, hostOperatorNamespace, memberOperatorNamespace string) {
 	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
 		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
 			if obj.GetNamespace() == namespace {
-				obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
-					Conditions: []toolchainv1alpha1.Condition{
-						{
-							Type:   toolchainv1alpha1.ConditionReady,
-							Status: corev1.ConditionTrue,
-						},
-					},
-				}
+				_mockCreateToolchainClusterWithReadyConditionModify(t, obj, hostOperatorNamespace, memberOperatorNamespace)
 			}
 		}
 		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
-func mockCreateToolchainClusterWithReadyCondition(fakeClient *test.FakeClient) {
+func mockCreateToolchainClusterWithReadyCondition(t *testing.T, fakeClient *test.FakeClient, hostOperatorNamespace, memberOperatorNamespace string) {
 	fakeClient.MockCreate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.CreateOption) error {
 		if obj, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
-			obj.Status = toolchainv1alpha1.ToolchainClusterStatus{
-				Conditions: []toolchainv1alpha1.Condition{
-					{
-						Type:   toolchainv1alpha1.ConditionReady,
-						Status: corev1.ConditionTrue,
-					},
-				},
-			}
+			_mockCreateToolchainClusterWithReadyConditionModify(t, obj, hostOperatorNamespace, memberOperatorNamespace)
 		}
 		return fakeClient.Client.Create(ctx, obj, opts...)
 	}
 }
 
+func _mockCreateToolchainClusterWithReadyConditionModify(t *testing.T, obj *toolchainv1alpha1.ToolchainCluster, hostOperatorNamespace, memberOperatorNamespace string) {
+	var operatorNamespace string
+	if obj.GetNamespace() == hostOperatorNamespace {
+		operatorNamespace = memberOperatorNamespace
+	} else if obj.GetNamespace() == memberOperatorNamespace {
+		operatorNamespace = hostOperatorNamespace
+	} else {
+		assert.Fail(t, "the mock create of ToolchainCluster only works in host operator namespace %s or member operator namespace %s but the creation of toolchain cluster was requested in %s", hostOperatorNamespace, memberOperatorNamespace, obj.GetNamespace())
+	}
```

Comment:
simplified version:
```suggestion
	require.Contains(t, []string{hostOperatorNamespace, memberOperatorNamespace}, obj.GetNamespace())
	operatorNamespace := hostOperatorNamespace
	if obj.GetNamespace() == hostOperatorNamespace {
		operatorNamespace = memberOperatorNamespace
	}
```

Reply:
I would like to keep the explicit fail there so that we discover wrong usage of this helper. Not a big deal, but I think error conditions like that helps with code comprehension.